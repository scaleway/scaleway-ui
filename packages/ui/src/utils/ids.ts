const chr4 = () => Math.random().toString(16).slice(-4)
const uuid = () => Array.from({ length: 8 }, chr4).join('')

export const getUUID = (prefix = ''): string =>
  `${prefix ? `${prefix}-` : ''}${uuid()}`
