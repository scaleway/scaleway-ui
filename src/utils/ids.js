import { useMemo } from 'react'

const chr4 = () => Math.random().toString(16).slice(-4)
const uuid = () => Array.from({ length: 8 }, chr4).join('')

export const getUUID = (prefix = '') => `${prefix ? `${prefix}-` : ''}${uuid()}`
export const useUUID = (prefix = '') => useMemo(() => getUUID(prefix), [prefix])
