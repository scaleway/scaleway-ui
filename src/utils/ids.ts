import { useMemo } from 'react'

const chr4 = () => Math.random().toString(16).slice(-4)
const uuid = () => Array.from({ length: 8 }, chr4).join('')

export const getUUID = (prefix = ''): string =>
  `${prefix ? `${prefix}-` : ''}${uuid()}`
export const useUUID = (prefix = ''): string =>
  useMemo(() => getUUID(prefix), [prefix])
