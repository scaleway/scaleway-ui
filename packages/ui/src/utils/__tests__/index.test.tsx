import { describe, expect, test } from '@jest/globals'
import { getUUID } from '../ids'

describe('ids', () => {
  describe('getUUID', () => {
    test('returns correctly without arguments', () => {
      expect(getUUID()).toMatchSnapshot()
    })

    test('returns correctly with a prefix', () => {
      expect(getUUID('ah')).toMatchSnapshot()
    })
  })
})
