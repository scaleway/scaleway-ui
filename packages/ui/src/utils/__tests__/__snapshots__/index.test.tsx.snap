// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ids > getUUID > returns correctly with a prefix 1`] = `"ah-6c756c756c756c756c756c756c756c75"`;

exports[`ids > getUUID > returns correctly without arguments 1`] = `"6c756c756c756c756c756c756c756c75"`;
